src/core/state.go

```diff
@@ -43,6 +43,12 @@ type TestTask struct {
 	Run   int
 }
 
+// Debug is the type for debugging a target
+type Debug struct {
+	Debugger string
```

Comment:
How do we expect this to be used? Will the build rules know about each debugger that we might configure, and how to invoke them? It doesn't seem like we have a use case for this yet so maybe we shouldn't add it until we have a better idea of that use case?

Reply:
We (the user) don't have to configure anything. The build rule owner however might want to provide different ways (i.e. debuggers) to debug the targets. Ideally, running, say, `plz help go_binary` the comment would include the keys to different debuggers that might be supported. If only one, the default one, is supported no need to make this visible.

If I'm authoring `go_binary` to support both Delve and GDB, I should take into account `CONFIG.DEBUG.DEBUGGER` that the use might have chosen.